src/widgets/BackButton.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React from 'react';
+import { StyleSheet, Text, TouchableOpacity } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { debounce } from '../utilities';
+
+import { getRouteTitle, RootNavigator } from '../navigation';
+
+import { goBack } from '../navigation/actions';
+
+import { BackIcon } from './icons';
+import { FlexRow } from './FlexRow';
+
+import { textStyles } from '../globalStyles';
+
+const mapStateToProps = state => {
+  const { pages = {} } = state;
+  const { currentRoute = '' } = pages;
+  const { [currentRoute]: currentPageState = {} } = pages;
+  const { pageObject } = currentPageState;
+
+  return { title: getRouteTitle(pageObject, currentRoute) };
```

Comment:
Hmm, I think this is elegant and concise, but I feel like "error handling" by deferring to passing `null` to `getRouteTitle` is a bit confusing. You might think OTT, so feel free to disagree, but something like this has more explicit branches which would make debugging/testing easier:

```suggestion
  const defaultProps = { title: '' };
  const { pages = {} } = state;
  const { currentRoute } = pages;
  if (!currentRoute) return defaultProps;
  const { [currentRoute]: currentPageState } = pages;
  if (!currentPageState) return defaultProps;
  const { pageObject } = currentPageState;
  if (!pageObject) return defaultProps;
  return { ...defaultProps, title:  getRouteTitle(pageObject, currentRoute) };
```

Reply:
this definitely seems odd and not to your regular "one path no early returns" arguments ! hehhee

Reply:
Your problem is with calling `getRouteTitle` with null so wouldn't something like:

`return {title: currentRoute ? getRouteTitle(.. ) : ""` address it?

Reply:
As above, just my personal option but:

My thought was more that unless it's a pattern throughout the codebase, it's probably more consistent to prevent passing `null` parameters in the caller, rather than assuming the function will handle `null` parameters and return a sensible value. If I didn't know the contract that `getRouteTitle` enforces, I have to double check that `getRouteTitle` behaves sensibly when it is called with basically no parameters. IMO this is the sort of thing that can cause bugs and something to be careful of when function params aren't typed.